Update doc reference

aowen87 opened the pull request:
### Description

Resolves #3766 

I've replaced the old documentation in VisIt's Help menu with our sphinx documentation. I've updated the QvisHelpWindow to reflect these changes, deleted all of the old html help files, and added a build of our sphinx manual to src/resources/help/en_US/manual.

**Note**: the documentation will need to be manually rebuilt before each release ensure it is up-to-date. In the future, we will likely automate this process. 


### Type of change

Please select one below (*Please click check boxes AFTER submitting ticket*)

- [ ] Bug fix
- [x] New feature
- [x] New Documentation
- [ ] Other (please describe below)

### How Has This Been Tested?
I've looked through the help menu to make sure the manual translated over to our GUI.

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the release notes
- [ ] I have made corresponding changes to the documentation
- [ ] I have added debugging support to my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added any new baselines to the repo
- [x] I have assigned reviewers (see [VisIt's PR procedures](https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/pr_create.html#reviewers) for more information).

aowen87 committed:
experimenting

aowen87 committed:
adding sphinx manual to help page

aowen87 committed:
adding the manual pages

aowen87 committed:
deleting old manual pages

aowen87 committed:
comments and cleanup

aowen87 committed:
Merge branch 'develop' into feature/maguire7/2019_10_update_doc_reference

aowen87 committed:
removing extra whitespace

brugger1 reviewed (commented):
It looks like you are duplicating all the manual files under resources/help/en_US/manual. Is that correct? If so, can we avoid doing that, since that could increase the size of our repo considerably.

markcmiller86 commented:
> It looks like you are duplicating all the manual files under resources/help/en_US/manual. Is that correct? If so, can we avoid doing that, since that could increase the size of our repo considerably.

FWIW...I agree with @brugger1 here. We should be seeking an approach where we don't duplicate content in our repo but instead *generate* on demand the various needed flavors of that content as part of our *release* processes.

aowen87 commented:
> It looks like you are duplicating all the manual files under resources/help/en_US/manual. Is that correct? If so, can we avoid doing that, since that could increase the size of our repo considerably.

@brugger1 is the documentation already built somewhere in the repo? I was having a hard time finding this out. We have a lot of documentation in the visit/docs directory, but I didn't see the sphinx generated documentation.

@markcmiller86 Yes, this is what I was asking you about on the issue page. It sounds like you already have a method for updating these docs for the live browser pages. Can we leverage this for the releases as well?

aowen87 commented:
If we don't want to have the documentation built in the repo, we can just run the documentation build script while building a release. This can be done manually, or, if there is a script that we use for building releases, I can add the command to this script.

markcmiller86 commented:
> @markcmiller86 Yes, this is what I was asking you about on the issue page. It sounds like you already have a method for updating these docs for the live browser pages. Can we leverage this for the releases as well?

Sorry. I hadn't picked up on that. So, the short answer is not really. I mean, ultimately, what needs to happen is that the build commands for the Sphinx docs have to get executed. However, the stuff you currently see on ReadTheDocs is managed much like other *services* are on GitHub like Travis CI for example. Some lite-weight VM is launched somewhere in the cloud with all the tools necessary to run `sphinx-build`, the html (and PDF) is generated and the results get *hosted*. ReadTheDocs is basically providing all that service for free. It gets tiggered automatically by a *web hook* every time there is an update to the `docs` dir on our `develop` branch. The newly generated content tends to go live ~5-10 mins thereafter. I can manually configure ReadTheDocs to build other branches of development too.  I have done this occasionally when I know someone has been doing large changes to docs and they'd like to see it hosted on RTD before a final merge to develop.

Now, it is certainly possible for us to somehow go download the HTML files from wherever RTD is hosting them and stick them in the right place when preparing a release. Would that be easier/better than just generating them locally when need to prepare a release? I don't think so. OTOH, it might result in more constistent look-and-feel of how the docs appear within VisIt's help system vs. how they appear hosted on RTD.

aowen87 commented:
@markcmiller86 thanks, Mark. I sent you an email after talking with Eric a few minutes ago as well, but I believe this has answered most of my questions. 
Moving forward, I'm going to look into how difficult it would be to install sphinx-build within VisIt's python and have our build system generate the documentation when running make (and making sure it's not rebuilt every time...).

markcmiller86 commented:
> @markcmiller86 thanks, Mark. I sent you an email after talking with Eric a few minutes ago as well, but I believe this has answered most of my questions.
> Moving forward, I'm going to look into how difficult it would be to install sphinx-build within VisIt's python and have our build system generate the documentation when running make (and making sure it's not rebuilt every time...).

I wonder if it might be worth making this an *optional* rather than a *required* aspect to running cmake in preparation for a build of VisIt? I don't think, for example, it should be a default/required part of a build, especially if being done via `build_visit`. Only reason I say that is to avoid adding more hurdles to building VisIt for the non-experts.

markcmiller86 commented:
FWIW...while poking around a bit, I happened to notice [this line](https://github.com/visit-dav/visit/blob/653ad3f4493935e1de2f21f92726b5d9327710d5/src/doc/conf.py#L64) and I believe it is causing RTD to indicate titles on all our pages (look at the your browser's tab) to indicate the docs are for 2.11.

Keeping that darn number up to date seems like a pain. I wonder if we should either leave that blank in `conf.py` or maybe set it on the sphinx-build command line we issue from CMake-ification rather than have it hard-coded into this file.

aowen87 commented:
> FWIW...while poking around a bit, I happened to notice [this line](https://github.com/visit-dav/visit/blob/653ad3f4493935e1de2f21f92726b5d9327710d5/src/doc/conf.py#L64) and I believe it is causing RTD to indicate titles on all our pages (look at the your browser's tab) to indicate the docs are for 2.11.
> 
> Keeping that darn number up to date seems like a pain. I wonder if we should either leave that blank in `conf.py` or maybe set it on the sphinx-build command line we issue from CMake-ification rather than have it hard-coded into this file.

Hmm... that's a good question. There must be a better way than hardcoding it in and having to remember to update it constantly.

aowen87 committed:
removing manual from repo

aowen87 commented:
It appears that building python sphinx from source requires python3.5 or greater, even though it is compatible with python2.7 and up. Unless I can find another way around this, having sphinx within VisIt's python might not be an option.

aowen87 commented:
I was able to get VisIt to install python3 and use this to build sphinx, but the newer version (2.2) appears to be incompatible with our documentation. We're apparently using sphinx 1.3. 
I found some older versions of sphinx to try out instead, and they apparently don't require having python3 for the installation process. I'm trying one out now.

aowen87 committed:
add sphinx to our python and build the docs with cmake

aowen87 commented:
I've added in some changes to install the sphinx package and have the documentation built during CMake. This has worked within contained tests, but I need to run all of build_visit to make sure there aren't any unforeseen side effects.

aowen87 commented:
I'm running into issues with getting mac to build the sphinx package. With these older versions at least, I'm running into this error during sphinx installation:
`[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:661) -- Some packages may not be found!`

It looks like it's going to pypi in search of package dependencies and ends up not being able to find them. I'm not sure if this is because we are using such an old version or if it's something else.

aowen87 committed:
build the newest sphinx with python3

aowen87 committed:
get the rtd theme working

aowen87 commented:
Installing the old version of sphinx from source seems to be a lost cause. 

I reverted back to installing python3 with the newest sphinx. The issues I was seeing earlier came down to the theme we were using (an old version of rtd). I added the latest version of the rtd theme as an installation dependency for sphinx, removed our old themes directory, and this seems to have resolved all issues. 

I'm currently running a full build to make sure everything works smoothly.

aowen87 commented:
Putting this merge off until after 3.1 release.
